pkg/apiserver/apiserver.go

```diff
@@ -127,6 +128,35 @@ func runE(flags *flags, log logr.Logger) error {
 		return fmt.Errorf("creating client: %w", err)
 	}
 
+	// Set up cache for account
+	accountCache, err := cache.New(cfg, cache.Options{
+		Scheme: scheme,
+		Mapper: mapper,
+	})
+	if err != nil {
+		return fmt.Errorf("creating cache for account: %w", err)
+	}
+	if err := v1.RegisterAccountUsernameFieldIndex(accountCache); err != nil {
+		return fmt.Errorf("fail to register field index for Account Username: %w", err)
+	}
+	accountClient := &client.DelegatingClient{
+		Reader:       accountCache,
+		Writer:       c,
+		StatusClient: c,
+	}
+	log.Info("start cache")
+	go func() {
+		if err := accountCache.Start(ctrl.SetupSignalHandler()); err != nil {
```

Comment:
Let's have main scoped context, and not create new one at each turn. (Replacing both `context.Background()` as well as `ctrl.SetupSignalHandler()` )

Reply:
done

Reply:
I had something else in mind; something like this: 

```
go func() {
   <-ctrl.SetupSignalHandler()
   cancel()
}
```

and using the `ctx.Done()` when someone requests the stop chan.